core/app/presenters/spree/product_summary_presenter.rb

```diff
@@ -18,7 +18,7 @@ def call
     def images
       @product.images.map do |image|
         {
-          url_product: url_for(image.url(:product))
+          url_product: rails_representation_url(image.url(:product), only_path: true)
```

Comment:
Since the `#url` method returns instance of `ActiveStorage::Variant` we have to use `#rails_representation_url` here. See https://github.com/rails/rails/issues/32500#issuecomment-380004250 for details.